cartoframes/dataset.py

```diff
@@ -24,51 +33,74 @@ class Dataset(object):
     PUBLIC = 'public'
     LINK = 'link'
 
+    TABLE_TYPE = 'Table'
+    QUERY_TYPE = 'Query'
+    DATAFRAME_TYPE = 'DataFrame'
+    GEODATAFRAME_TYPE = 'GeoDataFrame'
+
     DEFAULT_RETRY_TIMES = 3
 
-    def __init__(self, carto_context, table_name, schema='public', df=None):
-        self.cc = carto_context
-        self.table_name = normalize_name(table_name)
+    def __init__(self, type, data, context=None, schema='public', table_name=None):
+        self.type = type
+        self.data = data
```

Comment:
After reading the whole class I was a bit confusing with this variable name, some times it's a query, other times a table_name, other times a dataframe, etc.

Also we expose it outside in different flavours via the `get_data` method.

I'd suggest to reconsider the implementation of this class in general.

Reply:
Yes, we are revisiting this class and these properties